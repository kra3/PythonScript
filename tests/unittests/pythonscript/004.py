def telekin(self):
    print self.name, 'is telekinesing'


def higher_level_power(class_name, parents, attrs):
    attrs.telekin = telekin
    return type(class_name, parents, attrs)


class Person:

    __metaclass__ = higher_level_power

    def __init__(self, name):
        self.name = name

    def walk(self):
        print self.name, 'is walking'


aria = Person('aria')
aria.walk()
aria.telekin()
