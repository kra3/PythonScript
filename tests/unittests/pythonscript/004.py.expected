aria is walking
aria is telekinesing
